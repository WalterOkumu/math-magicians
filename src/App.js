import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Navbar from './components/Navbar';
import CalculatorPage from './pages/CalculatorPage';
import HomePage from './pages/HomePage';
import QuotePage from './pages/QuotePage';

const App = () => (
  <>
    <Navbar />
    <Routes>
      <Route path="/" element={<HomePage />} />
      <Route path="/calculator" element={<CalculatorPage />} />
      <Route path="/quote" element={<QuotePage />} />
    </Routes>
  </>
);

export default App;
