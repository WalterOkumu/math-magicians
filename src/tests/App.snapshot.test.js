import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';

describe('Jest snapshot to test components', () => {
  test('Matches DOM snapshot', () => {
    const domTree = renderer
      .create(
        <React.StrictMode>
          <Router>
            <App />
          </Router>
        </React.StrictMode>,
      )
      .toJSON();

    expect(domTree).toMatchSnapshot();
  });
});
