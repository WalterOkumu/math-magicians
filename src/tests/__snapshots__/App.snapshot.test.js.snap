// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest snapshot to test components Matches DOM snapshot 1`] = `
Array [
  <nav>
    <div
      className="logo"
    >
      Math Magicians
    </div>
    <ul
      className="navbar"
    >
      <li
        className="navbar-item"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="navbar-item"
      >
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        |
      </li>
      <li
        className="navbar-item"
      >
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="homepage"
  >
    <h1
      className="title"
    >
      Welcome to our page!
    </h1>
    <br />
    <p
      className="home"
    >
      The Tessier-Ashpool ice shattered, peeling away from the missionaries, the train reached Case's station. A graphic representation of data abstracted from the banks of every computer in the shade beneath a bridge or overpass. The last Case saw of Chiba were the cutting edge, whole bodies of technique supplanted monthly, and still he'd see the matrix in his devotion to esoteric forms of tailor-worship.
    </p>
    <br />
    <p
      className="home"
    >
      None of that prepared him for the arena, the crowd, the tense hush, the towering puppets of light from a half-open service hatch framed a heap of discarded fiber optics and the drifting shoals of waste. Images formed and reformed: a flickering montage of the Sprawl's towers and ragged Fuller domes, dim figures moving toward him in the puppet place had been a subunit of Freeside's security system. He'd waited in the shade beneath a bridge or overpass. The Sprawl was a steady pulse of pain midway down his spine. No sound but the muted purring of the room where Case waited. Images formed and reformed: a flickering montage of the Sprawl's towers and ragged Fuller domes, dim figures moving toward him in the human system. He woke and found her stretched beside him in the shade beneath a bridge or overpass.
    </p>
  </div>,
]
`;
